src/Common/src/System/Net/Security/NegotiateStreamPal.Unix.cs

```diff
@@ -165,6 +166,70 @@ internal static string QueryContextAuthenticationPackage(SafeDeleteContext secur
             return status == Interop.NetSecurityNative.Status.GSS_S_COMPLETE;
         }
 
+        private static bool GssAcceptSecurityContext(
+            ref SafeGssContextHandle context,
+            byte[] buffer,
+            out byte[] outputBuffer,
+            out uint outFlags)
+        {
+            if (context == null)
+            {
+                context = new SafeGssContextHandle();
+            }
+
+            Interop.NetSecurityNative.GssBuffer token = default(Interop.NetSecurityNative.GssBuffer);
+            Interop.NetSecurityNative.Status status;
+
+            try
+            {
+                Interop.NetSecurityNative.Status minorStatus;
+                status = Interop.NetSecurityNative.AcceptSecContext(out minorStatus,
+                                                                    ref context,
+                                                                    buffer,
+                                                                    buffer?.Length ?? 0,
+                                                                    ref token,
+                                                                    out outFlags);
+
+                if ((status != Interop.NetSecurityNative.Status.GSS_S_COMPLETE) &&
+                    (status != Interop.NetSecurityNative.Status.GSS_S_CONTINUE_NEEDED))
+                {
+                    throw new Interop.NetSecurityNative.GssApiException(status, minorStatus);
```

Comment:
If this throws, what happens to the SafeGssContextHandle we created and stored into `context` earlier (if null was passed in)?  Will that be left for finalization, and if so, could we instead Dispose of it (presumably only if we were the one to create it here)?

Reply:
Hmm, that applies to the existing EstablishSecurityContext as well.